tools/apigee-envoy-quickstart/aekitctl.sh

```diff
@@ -0,0 +1,149 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+usage() {
+    echo -e "$*\n usage: $(basename "$0")" \
+        "-t <type> -a <action>\n" \
+        "example: $(basename "$0") -t istio-apigee-envoy -a install\n" \
+        "example: $(basename "$0") -t istio-apigee-envoy -a delete\n" \
+        "example: $(basename "$0") -t standalone-apigee-envoy -a install\n" \
+        "example: $(basename "$0") -t standalone-apigee-envoy -a delete\n" \
+        "Parameters:\n" \
+        "-t --type        : Apigee protected Envoy installation type, valida values 'istio-apigee-envoy'. 'standalone-apigee-envoy'\n" \
+        "-a --action      : Install or Delete action, valid values 'install', 'delete'\n" \
+        "-p --platform    : Standalone install for platform opdk or edge, valid values 'opdk', 'edge'\n"
+    exit 1
+}
+
+init() {
+    export CLUSTER_CTX="gke_${PROJECT_ID}_${CLUSTER_LOCATION}_${CLUSTER_NAME}"
+    export ENVOY_AX_SA="x-apigee-envoy-adapter-2-sa"
+    export CLI_HOME=$ENVOY_HOME/apigee-remote-service-cli
+    export REMOTE_SERVICE_HOME=$ENVOY_HOME/apigee-remote-service-envoy
+    export ENVOY_CONFIGS_HOME=$CLI_HOME/envoy-configs-and-samples
+    export AX_SERVICE_ACCOUNT=$ENVOY_HOME/$ENVOY_AX_SA.json
+    export NAMESPACE="apigee"
+    if [ "$PLATFORM" == 'edge' ]; then
+        export MGMT_HOST="https://api.enterprise.apigee.com"
+    fi
+}
+
+createDir() {
+
+    mkdir $CLI_HOME
+    mkdir $REMOTE_SERVICE_HOME
+
+}
+
+PARAMETERS=()
+
+if [[ $# -ne 4 && $# -ne 6 ]]
+then
+   usage;
+   exit 1;
+fi  
+
+while [[ $# -gt 0 ]]
+do
+    param="$1"
+    case $param in
+        -t|--type)
+        export INSTALL_TYPE="$2"
+        shift
+        shift
+        ;;
+        -a|--action)
+        export ACTION="$2"
+        shift
+        shift
+        ;;
+        -p|--platform)
+        export PLATFORM="$2"
+        shift
+        shift
+        ;;
+        *)
+        PARAMETERS+=("$1")
+        shift
+        ;;
+    esac
+done
+ 
+if [[ -z $INSTALL_TYPE ]]; then
+    usage "installation type is a mandatory field"
+fi
+
+if [[ -z $ACTION ]]; then
+    usage "action is a mandatory field"
+fi
+
+if [ "$PLATFORM" != 'opdk' ] && [ "$PLATFORM" != 'edge' ]
+then
+    gke-gcloud-auth-plugin --version 2>&1 >/dev/null
+    RESULT=$?
+    if [ $RESULT -eq 0 ]; then
+        export USE_GKE_GCLOUD_AUTH_PLUGIN=True
+    fi
+fi
+
+init;
+
+if [ "$PLATFORM" == 'opdk' ] || [ "$PLATFORM" == 'edge' ]
+then
+    export TOKEN=$(echo -n "$APIGEE_USER":"$APIGEE_PASS" | base64 | tr -d \\r)
+    export TOKEN_TYPE="Basic"
+else
+    export TOKEN_TYPE="Bearer"
+    export MGMT_HOST="https://apigee.googleapis.com"
+fi
+
+./scripts/validate.sh
+
+if [ "$PLATFORM" == 'opdk' ] || [ "$PLATFORM" == 'edge' ]
+then
+    ./scripts/validate-opdk-edge-setup.sh
```

Comment:
I see a lot of files being pulled in on-the-fly here. I don't feel too strongly about it, but can't help think it would be better to keep these all in one place (i.e. at the top). I'll let others chime in with their thoughts on this...